src/masternode.cpp

```diff
@@ -870,6 +870,18 @@ bool CMasternodePing::CheckAndUpdate(CMasternode* pmn, int& nDos)
         return false;
     }
 
+    if(!fFromNewBroadcast) {
+        if (pmn->IsUpdateRequired()) {
+            LogPrint("masternode", "CMasternodePing::CheckAndUpdate -- masternode protocol is outdated, masternode=%s\n", vin.prevout.ToStringShort());
+            return false;
+        }
+
+        if (pmn->IsNewStartRequired()) {
+            LogPrint("masternode", "CMasternodePing::CheckAndUpdate -- masternode is completely expired, new start is required, masternode=%s\n", vin.prevout.ToStringShort());
+            return false;
```

Comment:
How are nodes in NEW_START_REQUIRED recovered if false is returned for new broadcasts ?